src/lib/Preview.js

```diff
@@ -541,15 +543,25 @@ class Preview extends EventEmitter {
                 params,
             );
 
-            this.api.reachability.downloadWithReachabilityCheck(downloadUrl);
+            downloadPromise = this.api.reachability.downloadWithReachabilityCheck(downloadUrl);
```

Comment:
It doesn't look like `downloadWithReachabilityCheck` actually returns a promise?